include/seqan3/search/algorithm/configuration/mode.hpp

```diff
@@ -0,0 +1,153 @@
+// ============================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ============================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides the mode configuration to define the search modes "all", "all_best", "best" and "strata".
+ * \author Christopher Pockrandt <christopher.pockrandt AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/algorithm/all.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/search/algorithm/configuration/utility.hpp>
+
+/*!\addtogroup search
+ * \{
+ */
+
+namespace seqan3::detail
+{
+
+struct search_mode_all {};
+
+struct search_mode_all_best {};
+
+struct search_mode_best {};
+
+} // namespace seqan3::detail
+
+namespace seqan3::search_cfg
+{
+
+//!\brief Configuration element to receive all hits within the error bounds.
+inline detail::search_mode_all constexpr all;
```

Comment:
`inline constexpr type var-name`

Reply:
I will refer for this discussion to @h-2 :stuck_out_tongue:

Long story short. constexpr should be at the same position as const and is going to be changed everywhere in the library soon (except for constexpr functions).

Reply:
😭 Then I am more for putting the const/constexpr left! 
It spells static constexpr type
Not static type constexpr

Reply:
No! We have always used east-const and it is the more consistent notation!! See http://slashslash.info/eastconst/ for example.
We just didn't follow through with `constexpr` so we need to fix that.

Reply:
Well, if you are so about consistency, why not following cpp core guidelines. I don't think the state explicitly, but every example is right const/constexpr